lib/mongo_ecto/encoder.ex

```diff
@@ -46,6 +46,8 @@ defmodule Mongo.Ecto.Encoder do
       :error       -> :error
     end
   end
+  def encode(%Mongo.Ecto.DiffEmbed{} = diff, _pk),
```

Comment:
Yeah, it's a great point. And I would to go through the list only once.

The way why I do this like that, is because I wanted initially to issue `$unset` queries on delete, instead of nullifying values (I still want to do that, but I decided to go one step at a time) - and I need to differentiate between `$set` and `$unset` in the normalized query when building the command.